checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -776,9 +788,38 @@ public void visitSwitchExpression17(Tree switchExprTree) {
             checkForNullability(
                     SwitchExpressionUtils.getExpression(switchExprTree), SWITCHING_NULLABLE);
         }
+        if (lintredundantNullComparison) {
+            ExpressionTree expression = SwitchExpressionUtils.getExpression(switchExprTree);
+            List<? extends CaseTree> cases = SwitchExpressionUtils.getCases(switchExprTree);
+            checkSwitchNullRedundant(expression, cases);
+        }
         super.visitSwitchExpression17(switchExprTree);
     }
 
+    /**
+     * Reports an error if the switch expression is @NonNull and the case expression is null
+     * literal.
+     *
+     * @param expression the switch expression
+     * @param cases the case expressions
+     */
+    private void checkSwitchNullRedundant(
+            ExpressionTree expression, List<? extends CaseTree> cases) {
+        AnnotatedTypeMirror switchType = atypeFactory.getAnnotatedType(expression);
+        for (CaseTree caseTree : cases) {
+            List<? extends ExpressionTree> caseExpressions =
```

Comment:
This only iterates over the case expressions, whereas `isEnhancedSwitchStatement` only iterates over the case labels.
Which one is right? Or should both iterate over both the case expressions & labels?

Reply:
According to java doc, they are pretty much the same except for the type(`getexpression` returns expression tree and `getlabels` returns tree. I have tested both and works ok. Let's use getlabels to keep consistent.